forage-android/src/main/java/com/joinforage/forage/android/collect/BTPinCollector.kt

```diff
@@ -32,56 +33,36 @@ internal class BTPinCollector(
         return vaultType
     }
 
-    override suspend fun collectPinForBalanceCheck(
+    override suspend fun submitBalanceCheck(
         paymentMethodRef: String,
         cardToken: String,
         encryptionKey: String
     ): ForageApiResponse<String> {
+        val logAttributes = mapOf(
+            "merchant_ref" to merchantAccount,
+            "payment_method_ref" to paymentMethodRef
+        )
+
         // If the PIN isn't valid (less than 4 numbers) then return a response here.
         if (!pinForageEditText.getElementState().isComplete) {
-            logger.w(
-                "[BT] User attempted to submit an invalid PIN",
-                attributes = mapOf(
-                    "merchant_ref" to merchantAccount,
-                    "payment_method_ref" to paymentMethodRef
-                )
-            )
-            return ForageApiResponse.Failure(
-                ForageConstants.ErrorResponseObjects.INCOMPLETE_PIN_ERROR
-            )
+            return returnIncompletePinError(logAttributes, logger)
```

Comment:
Nice opportunity to DRY up the code! In the future we can even write tests against these log messages to avoid accidental typos